### PR TITLE
Improve Security of Barcode Retrieval from PDF

### PR DESCRIPTION
**Pull request by dimitriospapathanasiou**
In past issues, it was expressed that security could be improved regarding the PDF process. I have made some minor changes in the Util file, as a start.

These changes improve resource management and error handling while maintaining good coding practices. Firstly, I closed the resources in a finally block, to ensure they are properly released. I also made a check to validate Uri before usage, which could potentially have some additional validation in a later commit, to ensure it points to a valid PDF file. Additional minor changes are performed to handle errors and sensitive information more securely. I have performed tests to see if there is some deviation of these changes from the previous functionalities, and all appeared to be clear.

### CODE REVIEW

File: app/src/main/java/protect/card_locker/Utils.java [hunk 1]
```diff
@@ -169,48 +169,62 @@ static public List<BarcodeValues> retrieveBarcodesFromImage(Context context, Uri
 
     static public List<BarcodeValues> retrieveBarcodesFromPdf(Context context, Uri uri) {
         Log.i(TAG, "Received PDF file with possible barcode");
-
+
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/Utils.java:172]: Unnecessary whitespace

```suggestion

```

File: app/src/main/java/protect/card_locker/Utils.java [hunk 2]
```diff
@@ -169,48 +169,62 @@ static public List<BarcodeValues> retrieveBarcodesFromImage(Context context, Uri
...
         }
-        renderer.close();
-
+    
         if (barcodesFromPdfPages.isEmpty()) {
             Log.i(TAG, "No barcode found in pdf file");
             Toast.makeText(context, R.string.noBarcodeFound, Toast.LENGTH_LONG).show();
         }
-
+
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/Utils.java:224]: Unnecessary whitespace

```suggestion

```

File: app/src/main/java/protect/card_locker/Utils.java [hunk 3]
```diff
@@ -169,48 +169,62 @@ static public List<BarcodeValues> retrieveBarcodesFromImage(Context context, Uri
...
+    
         if (barcodesFromPdfPages.isEmpty()) {
             Log.i(TAG, "No barcode found in pdf file");
             Toast.makeText(context, R.string.noBarcodeFound, Toast.LENGTH_LONG).show();
         }
-
+    
         return barcodesFromPdfPages;
     }
+
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/Utils.java:227]: Unnecessary whitespace

```suggestion
```

**Review by TheLastProject — CHANGES_REQUESTED** [review]
I don't really agree this PR improves the security, but cleaning up resources as soon as we don't need them anymore instead of waiting for garbage collection and having some more null checking to prevent crashing in edge cases is still a good improvement, thanks :+1: 

Just some unnecessary whitespace :)

**Comment by dimitriospapathanasiou** [comment]
Thank you for your review! I removed the unnecessary whitespace.
